Add definition of Omni Layer token

D4nte opened the pull request:
Editor: @bonomat 

Resolves #50

LLFourn commented on RFC-010-SWAP-Omni-Layer.md:
```diff
@@ -0,0 +1,117 @@
+This RFC specifies how Bitcoin based Omni Layer Assets MUST be described in [RFC002](./RFC-002-SWAP.md) SWAP messages.
+OmniLayer refers specifically to [Omni Protocol](https://github.com/OmniLayer/spec).
+As required by [RFC002](./RFC-002-SWAP.md), this RFC defines the *asset* value `omni_layer`.
+This in combination with subsequent RFCs that define concrete execution steps for particular SWAP protocols enable users to negotiate and execute Omni Layer Token swaps.
```
I'd prefer we follow the way I did ERC20:
1. Call the file RFC-010-Omni-Layer.md
2. Make the description stand alone without regards to SWAP messages (the asset definition could easily be used outside of SWAP messages). Just mention that:
```
The Asset type header was introduced in [RFC002](./RFC-002-SWAP.md) to describe assets being exchanged in a COMIT SWAP protocol.
```

LLFourn commented on RFC-010-SWAP-Omni-Layer.md:
```diff
@@ -0,0 +1,117 @@
+It also allow the definition of new tokens by the mean of a Bitcoin transaction containing specific data.
+
+This asset MUST be used on the Bitcoin Ledger, as defined in [RFC-004](./RFC-004-SWAP-Bitcoin.md).
+Subsequent RFCs MAY introduce the support of other ledgers for the Omni Layer token Asset.
```
Omni layer supports other ledgers?

bonomat replied:
Yes, there is an implementation for Litecoin and Thether actually planned moving to Litecoin sometime ago... not sure whether they did it though.

LLFourn commented on RFC-010-SWAP-Omni-Layer.md:
```diff
@@ -0,0 +1,117 @@
+
+For a full list of pre-defined coins refer to [Omni Layer Specs](https://github.com/OmniLayer/spec).
+ 
+To specify an Omni Layer Token in one of the [RFC002](./RFC-002-SWAP.md) asset headers (`alpha_asset` or `beta_asset`) use the value `omni_layer` with the following parameter:
```
Make this sentence not reference RFC002 like:
>To describe ERC20 asset in an Asset type header specify `erc20` as the value along with the following parameters:

LLFourn commented on RFC-010-SWAP-Omni-Layer.md:
```diff
@@ -0,0 +1,117 @@
+A given Omni Layer token is defined as *divisible* or *indivisible* at creation.
+To cater for these two types of token, the [Omni Protocol- Number of coins field](https://github.com/OmniLayer/spec#field-number-of-coins) defines the following behaviour in Omni Layer transactions:
+- If the token is **indivisible** then the `quantity` value is the exact number of tokens: `1` represents one token.
+- If the token is **divisible** then the `quantity` value is to be divided by 100,000,000: `100,000,000` represents one token.
```
```suggestion
- If the token is **divisible** then the `quantity` value represents a hundred millionth of a token: `100,000,000` represents one token.
```

LLFourn commented on RFC-010-SWAP-Omni-Layer.md:
```diff
@@ -0,0 +1,117 @@
+
+While *Omni* or *OmniCoin* holds a special place among the tokens as being predefined and having special properties in the Omni Protocol, it is not different than other Omni Layer tokens in the context of the COMIT protocol family.
+
+Each Omni Layer token defines its name.
```
I don't think this sentence is useful for an implementor. (I know I added it in ERC20 but I don't think it helps in retrospect).

D4nte commented:
@LLFourn I am sorry I added you as reviewer, I removed you after.
All your comments will be ignored until the Editor (@bonomat) has done a first pass.

bonomat commented:
@LLFourn : I'm done with proofreading -> This PR is open for comments now ;)

thomaseizinger commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,120 @@
+It also allows the definition of new assets by the mean of a Bitcoin transaction containing specific data.
+
+This asset MUST be used on the Bitcoin Ledger, as defined in [RFC-004](./RFC-004-SWAP-Bitcoin.md).
+Subsequent RFCs MAY introduce the support of other ledgers for Omni Layer Assets.
```
Is this even possible? Are Omni assets a thing on ledgers other than bitcoin?

thomaseizinger replied:
If not, then the hypothetical sentence could be removed.

bonomat replied:
See question from Lloyd above: there is an implementation for Litecoin and Tether once wanted to move to Litecoin. So theoretically this would be possible... not sure how likely it is though. 
https://github.com/OmniLayer/omnicore-lite

thomaseizinger commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,120 @@
+
+| Value        | Description      |
+:---           |---               |
+| `omni_layer` | Omni Layer Asset |
```
:wrench: What about calling this one `omni` instead of `omni_layer`? `omni_layer` sounds more like a ledger ...

LLFourn commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,120 @@
+Notable Omni assets have the following property ids:
+- Omni Coin (predefined): `1`
+- Test Omni Coin (predefined): `2`
+- TetherUS: `31`
```
As an implementer, it's not clear what I have to do with this information. Should we be listing in a table or all the allowed `property_id`s or is this like ERC20 where we negotiate based on a the contract address and don't care about particular currencies. If it's the latter I think remove this.

LLFourn commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,120 @@
+- Test Omni Coin (predefined): `2`
+- TetherUS: `31`
+
+For a full list of pre-defined Omni assets refer to [Omni Layer Specs](https://github.com/OmniLayer/spec).
```
Where is this list though?

LLFourn commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,112 @@
+Each Omni asset defines its name.
+Omni assets are uniquely identified by a *property id*.
+
+For a full list of pre-defined Omni assets refer to [Omni Layer Specs](https://github.com/OmniLayer/spec#field-currency-identifier).
```
```suggestion
For a list of pre-defined Omni assets refer to [Omni Layer Specs](https://github.com/OmniLayer/spec#field-currency-identifier).
```

Full sounds weird here.

LLFourn commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,112 @@
+A given Omni asset is defined as *divisible* or *indivisible* at creation.
+To cater for these two types of assets, the following behaviour is defined in the Omni Layer protocol field [Number of coins field](https://github.com/OmniLayer/spec#field-number-of-coins):
+- If the asset is **indivisible** then the `quantity` value is the exact number of assets: `1` represents one asset.
+- If the asset is **divisible** then the `quantity` value represents a hundred millionth of an asset: `100,000,000` represents one asset.
```
❓ can we use the term "token" here instead of asset because it kind of conflicts with our nomenclature

LLFourn commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,112 @@
+
+A given Omni asset is defined as *divisible* or *indivisible* at creation.
+To cater for these two types of assets, the following behaviour is defined in the Omni Layer protocol field [Number of coins field](https://github.com/OmniLayer/spec#field-number-of-coins):
+- If the asset is **indivisible** then the `quantity` value is the exact number of assets: `1` represents one asset.
```
```suggestion
- If the asset is **indivisible** then the `quantity` value is the exact number of tokens, i.e. `1` represents one token.
```

LLFourn commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,112 @@
+A given Omni asset is defined as *divisible* or *indivisible* at creation.
+To cater for these two types of assets, the following behaviour is defined in the Omni Layer protocol field [Number of coins field](https://github.com/OmniLayer/spec#field-number-of-coins):
+- If the asset is **indivisible** then the `quantity` value is the exact number of assets: `1` represents one asset.
+- If the asset is **divisible** then the `quantity` value represents a hundred millionth of an asset: `100,000,000` represents one asset.
```
```suggestion
- If the asset is **divisible**, then the `quantity` value represents a hundred millionth of a token, i.e. `100,000,000` represents one token.
```

LLFourn commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,112 @@
+
+The Asset type header was introduced in [RFC002](./RFC-002-SWAP.md) to describe assets being exchanged in a COMIT SWAP protocol.
+
+This in combination with subsequent RFCs that define concrete execution steps for particular SWAP protocols enables users to negotiate and execute Omni Layer asset swaps.
```
🔧 I'd prefer not to have this sentence. This RFC can just describe Omin layer assets and not promise what other RFCs are going to do.

D4nte replied:
```suggestion
```

D4nte commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,111 @@
+
+For a list of pre-defined Omni assets refer to [Omni Layer Specs](https://github.com/OmniLayer/spec#field-currency-identifier).
+
+To describe an Omni asset in an Asset type header specify `omni_layer` as the value along with the following paramters:
```
```suggestion
To describe an Omni asset in an Asset type header specify `omni` as the value along with the following paramters:
```

D4nte commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,111 @@
+
+## Registry extension
+
+This RFC extends the registry's [Assets table](./registry.md#assets) with the `omni_layer` Asset:
```
```suggestion
This RFC extends the registry's [Assets table](./registry.md#assets) with the `omni` Asset:
```

thomaseizinger commented on RFC-010-Omni-Layer.md:
```diff
@@ -0,0 +1,111 @@
+
+Note that property id is assigned by the Omni Layer Protocol at the creation of an asset - it is simply the next available property id.
+
+The property id MUST be of an existing asset on the target Bitcoin ledger.
```
:question: What do you mean by that? How can an implementer ensure that their implementation adheres to this `MUST`?

:x: I think this is a actual a case of an "unknown asset"-error that we can return as soon as #23 is finished!

D4nte replied:
Please make a suggestion ;)

thomaseizinger commented on registry.md:
```diff
@@ -100,6 +100,12 @@ And the possible parameters they each may have:
+
+| Parameter        | Value Type | Description                                            |
+|:-----------------|------------|--------------------------------------------------------|
+| `quantity`       | `u64`      | The amount of Omni Layer assets to be transferred |
```
:upside_down_face: 
```suggestion
| `quantity`       | `u64`      | The amount of Omni Layer assets to be transferred      |
```

thomaseizinger commented:
Blocked by #23 being merged in order to introduce a decline-reason/negotiation-error for: https://github.com/comit-network/RFCs/pull/51#discussion_r272867329

D4nte commented:
@bonomat please kindly review :)

D4nte commented:
Merged, it's a RFC so people can always decide to comment on the tracking issue that I forgot to open :).
